fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java

```diff
@@ -1772,7 +1772,7 @@ public void saveImage(ImageWriter imageWriter, File curFile) throws IOException
         LOG.info("start save image to {}. is ckpt: {}", curFile.getAbsolutePath(), GlobalStateMgr.isCheckpointThread());
 
         long saveImageStartTime = System.currentTimeMillis();
-        try (OutputStream outputStream = Files.newOutputStream(curFile.toPath())) {
+        try (FileOutputStream outputStream = new FileOutputStream(curFile)) {
```

Comment:
may create a new class `SyncFileOutputStream extends FileOutputStream`, hook the channel().force() in close() or flush().

Reply:
Autoclosing is convenient for programming, but its order is hard to guarantee. For example, in the code below, the file needs to be synced before `imageWriter.saveChecksum()`. If changed to autoclose, it would happen afterward. 
And more, in the StarMgrServer code, if SyncFileOutputStream is used, it becomes:
```java
try (FileOutputStream fos = new SyncFileOutputStream(ckpt);
     DataOutputStream dos = new DataOutputStream(fos)) {
    getStarMgr().dumpMeta(dos);
}
```
This could easily be modified by someone to:
```java
try (DataOutputStream dos = new DataOutputStream(new SyncFileOutputStream(ckpt))) {
    getStarMgr().dumpMeta(dos);
}
```
Which is dangerous
Since syncing is a critical operation, I believe it’s better to perform it explicitly.